### PR TITLE
AVL Tree implementation in C++

### LINKED ISSUE #2112: AVL tree
- [x] C
- [x] C++ - @Manish-cloud
- [ ] C# - @KanakalathaVemuru
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go
- [x] Java
- [ ] JS - @satvikchachra
- [ ] Kotlin
- [ ] Php - @chandrikadeb7
- [x] Python
- [x] Readme
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Manish-cloud**
Fixes #2112 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Adding implementation of AVL Tree
- 

#### Languages Used:
- C++
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Manish-cloud** [comment]
@raghav-dalmia  Kindly review my PR.

**Comment by Manish-cloud** [comment]
@MastersAbh Sir, I have removed extra file .

**Comment by MastersAbh** [comment]
I still see frequency of element file in python attached to this PR. Can you please remove that?

**Comment by Manish-cloud** [comment]
@MastersAbh Sir, please check my PR. I a have removed extra files and also deleted unwanted commits.

File: AVL_Tree/AVL_Tree.cpp [hunk 1]
```diff
@@ -0,0 +1,209 @@
+#include<iostream>
+using namespace std;
+
+struct tnode
+{
+    int data;
+    tnode * L,* R;
```
Comment (somya-kapoor) [inline AVL_Tree/AVL_Tree.cpp:7]: Add a space after ,

File: AVL_Tree/AVL_Tree.cpp [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    tnode()
+    {
+        L=R=NULL;
```
Comment (somya-kapoor) [inline AVL_Tree/AVL_Tree.cpp:10]: Add space around the operator

File: AVL_Tree/AVL_Tree.cpp [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        tnode * root;
+    public:
+        AVL(){root = NULL;}
+        tnode * LL(tnode *);
+        tnode * RR(tnode *);
+        tnode * LR(tnode *);
+        tnode * RL(tnode *);
+    
+        int height(tnode *);
+        int MAX(int,int);
```
Comment (somya-kapoor) [inline AVL_Tree/AVL_Tree.cpp:26]: Add space after ,

File: AVL_Tree/AVL_Tree.cpp [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        tnode * insert(tnode *,int);
+        void display();
+        void inorder(tnode *);
+        void preorder(tnode *);
+};
+
+tnode * AVL :: LL(tnode *parent)
+{
+    tnode *temp;
+    temp = parent->L;
```
Comment (somya-kapoor) [inline AVL_Tree/AVL_Tree.cpp:38]: Add space around ->

File: AVL_Tree/AVL_Tree.cpp [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+tnode * AVL :: RL(tnode *parent)
+{
+    parent->R = LL(parent->R);
+    parent = RR(parent);
+    return parent;
+}
+
+int AVL :: MAX(int a, int b)
+{
+    if(a>b)
```
Comment (somya-kapoor) [inline AVL_Tree/AVL_Tree.cpp:69]: Add space around the operator.
Do the required change in whole file

File: AVL_Tree/AVL_Tree.cpp [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    t.display();
+    cout<<endl;
+    return 0;
+}
+
+
+
+
+/*
+
```
Comment (somya-kapoor) [inline AVL_Tree/AVL_Tree.cpp:176]: Remove this line

File: AVL_Tree/AVL_Tree.cpp [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+int main()
+{
+    AVL t;
+    t.create();
+    t.display();
+    cout<<endl;
+    return 0;
+}
+
```
Comment (somya-kapoor) [inline AVL_Tree/AVL_Tree.cpp:171]: Remove this line

File: AVL_Tree/AVL_Tree.cpp [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+
```
Comment (somya-kapoor) [inline AVL_Tree/AVL_Tree.cpp:172]: Remove this line

File: AVL_Tree/AVL_Tree.cpp [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
```
Comment (somya-kapoor) [inline AVL_Tree/AVL_Tree.cpp:173]: Remove this line

File: AVL_Tree/AVL_Tree.cpp [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+    cout<<"\n Preorder : ";
+    preorder(root);
+}
+
+int main()
+{
+    AVL t;
+    t.create();
+    t.display();
+    cout<<endl;
```
Comment (somya-kapoor) [inline AVL_Tree/AVL_Tree.cpp:168]: Add space around the operator

**Comment by Manish-cloud** [comment]
@somya-kapoor Please review my PR. Added requested changes in file.

**Comment by Manish-cloud** [comment]
@MastersAbh @somya-kapoor  Completed Squash request.

**Pull request merged**